packages/pg-v5/commands/pull.js

```diff
@@ -125,19 +112,69 @@ const maybeTunnel = function * (herokuDb) {
   return herokuDb
 }
 
+function spawnPipe (commandOne, commandTwo) {
+  return new Promise((resolve, reject) => {
+    commandOne.stdout.on('data', (data) => {
+      commandTwo.stdin.write(data)
+    })
+
+    commandOne.stderr.on('data', (data) => {
+      console.log(`${data.toString().trim()}`)
+    })
+
+    commandOne.on('close', (code) => {
+      if (code !== 0) {
+        cli.exit(code)
+      }
+      commandTwo.stdin.end()
+    })
+
+    commandTwo.stderr.on('data', (data) => {
+      console.log(`${data.toString().trim()}`)
+    })
+
+    commandTwo.on('close', (code) => {
+      if (code === 0) {
+        resolve(undefined)
+      } else {
+        cli.exit(code)
+      }
+    })
+  })
+}
+
 const run = co.wrap(function * (sourceIn, targetIn, exclusions) {
   yield prepare(targetIn)
 
   const source = yield maybeTunnel(sourceIn)
   const target = yield maybeTunnel(targetIn)
-
   const exclude = exclusions.map(function (e) { return '--exclude-table-data=' + e }).join(' ')
 
-  let password = p => p ? ` PGPASSWORD="${p}"` : ''
-  let dump = `env${password(source.password)} PGSSLMODE=prefer pg_dump --verbose -F c -Z 0 ${exclude} ${connstring(source, true)}`
-  let restore = `env${password(target.password)} pg_restore --verbose --no-acl --no-owner ${connstring(target)}`
+  let dumpFlags = ['--verbose', '-F', 'c', '-Z', '0']
+  if (exclude !== '') { dumpFlags.push(`${exclude}`) }
+
+  dumpFlags = dumpFlags.concat(connstring(source, true).split(' '))
+
+  const dumpOptions = {
+    env: {
+      ...(source.password && { PGPASSWORD: `${source.password}` }),
```

Comment:
I think this could be cleaner:

```
const dumpOptions = {
  env: {
    PGSSLMODE: 'prefer'
  },
  encoding: 'utf8',
  shell: true
}
if (source.password) dumpOptions.PGPASSWOrD = source.password
```